pkg/lib/serverTime.js

```diff
@@ -321,13 +324,181 @@
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    function find_file_and_read(names, first) {
+        if (first === undefined)
+            first = 0;
+        if (first >= names.length)
+            return Promise.reject(new Error("Found none of " + names.join(",")));
+        return cockpit.file(names[first], { superuser: "try" }).read()
+                .then(text => {
+                    if (text)
+                        return [names[first], text];
+                    return find_file_and_read(names, first + 1);
+                });
+    }
+
+    function get_ntp_config_chronyd() {
+        return find_file_and_read(["/etc/chrony.conf", "/etc/chrony/chrony.conf"])
+                .then(([name, text]) => {
+                    const result = {
+                        supported: true,
+                        backend: "chronyd",
+                        file: name,
+                        standard: [],
+                        sources: [],
+                        extra: []
+                    };
+
+                    let in_cockpit_section = false;
+                    for (const line of text.split("\n")) {
+                        if (line == "# Added by Cockpit") {
+                            in_cockpit_section = true;
+                        } else {
+                            const fields = line.split(" ");
+                            if (fields[0] == "server") {
+                                if (in_cockpit_section) {
+                                    result.extra.push(fields[1]);
+                                } else {
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                }
+                            } else {
+                                in_cockpit_section = false;
+                                if (fields[0] == "pool")
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#pool" || fields[0] == "#server")
+                                    result.standard.push({ name: fields[1], enabled: false });
+                                if (fields[0] == "sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: false });
+                            }
+                        }
+                    }
+
+                    return result;
+                });
+    }
+
+    function set_ntp_config_chronyd(config) {
+        const chrony_file = cockpit.file(config.file, { superuser: "try" });
+
+        function confline(line, list, name) {
+            const c = list.find(e => e.name == name);
+            if (!c)
+                return line;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230124-110801-7b5801b2-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)